src/utils/Polygons.ts

```diff
@@ -11,6 +14,101 @@ import { svgPolygonToWKT,
 //
 // Please do not use the VectorFactory methods directly.
 export class Polygon {
+    public static add(a: Polygon, b: Polygon): Polygon {
+        if (a.empty) {
+            return b;
+        }
+        if (b.empty) {
+            return a;
+        }
+        const cpr = new ClipperLib.Clipper();
+        cpr.AddPaths(a.clipper, ClipperLib.PolyType.ptSubject, true);
+        cpr.AddPaths(b.clipper, ClipperLib.PolyType.ptClip, true);
+        const solutionPaths = new ClipperLib.Paths();
+
+        cpr.Execute(
+            ClipperLib.ClipType.ctUnion,
+            solutionPaths,
+            ClipperLib.PolyFillType.pftNonZero,
+            ClipperLib.PolyFillType.pftNonZero
+        );
+
+        const result = Polygon.fromClipper(solutionPaths);
+        return result;
+    }
+
+    public static subtract(a: Polygon, b: Polygon): Polygon {
+        if (a.empty || b.empty) {
+            return a;
+        }
+        const cpr = new ClipperLib.Clipper();
+        cpr.AddPaths(a.clipper, ClipperLib.PolyType.ptSubject, true);
+        cpr.AddPaths(b.clipper, ClipperLib.PolyType.ptClip, true);
+        const solutionPaths = new ClipperLib.Paths();
+
+        cpr.Execute(
+            ClipperLib.ClipType.ctDifference,
+            solutionPaths,
+            ClipperLib.PolyFillType.pftNonZero,
+            ClipperLib.PolyFillType.pftNonZero
+          );
+
+        const result = Polygon.fromClipper(solutionPaths);
+        return result;
+    }
+
+    public static intersect(a: Polygon, b: Polygon): Polygon {
+        if (a.empty || b.empty) {
+            return new Polygon();
+        }
+
+        const cpr = new ClipperLib.Clipper();
+        cpr.AddPaths(a.clipper, ClipperLib.PolyType.ptSubject, true);
+        cpr.AddPaths(b.clipper, ClipperLib.PolyType.ptClip, true);
+        const solutionPaths = new ClipperLib.Paths();
+
+        cpr.Execute(
+            ClipperLib.ClipType.ctIntersection,
+            solutionPaths,
+            ClipperLib.PolyFillType.pftNonZero,
+            ClipperLib.PolyFillType.pftNonZero
+          );
+
+        const result = Polygon.fromClipper(solutionPaths);
+        return result;
+    }
+
+    // Scale the polygon by a factor
+    public static scale(a: Polygon, factor: number): Polygon {
+        if (a.empty) {
+            return new Polygon();
+        }
+        if (factor === 1) {
+            return a;
+        }
+
+        // Match all the numbers in the SVG using a regex, which is faster than scanning in Javascript
+        const re = /\d+(\.\d*)?/g; // Adapted from here: https://stackoverflow.com/a/18085/871910
```

Comment:
We'll move it to a private static member, so it's only initialized once. This is definitely permature optimization, but it's so simple to do we'll just do it.

Reply:
As for scaling the polygon - we do it so that canvas is small. This was crucial for tablets. It did not occur to me to perform the scaling in SVG (although tablet GPUs are not strong, unless it's an iPad). Too bad. At this point I *really* don't want to change that, as it will probably require testing the Imaged Object Editor all over again, and fixing zooming and rotating. So let's worry about this if we see a performance issue.

Reply:
I understand now, you need to scale it draw it at the right size on the (shrunken) canvas.  Indeed, if it runs fine, then we should just leave it.